src/components/DateInput.jsx

```diff
@@ -217,7 +221,7 @@ class DateInput extends React.PureComponent {
             focused && styles.DateInput_input__focused,
             disabled && styles.DateInput_input__disabled,
           )}
-          aria-label={placeholder}
+          aria-label={!ariaLabelledBy && placeholder}
```

Comment:
i think you may want a ternary here, to ensure it gets set to undefined and not false?